utils/sl/setup_local.sh

```diff
@@ -43,79 +44,117 @@
 # SL_*PREFIX variables to be used by component specific wrapper scripts
 # to setup things like PYTHONPATH, CPATH, LIBRARY_PATH, as needed.
 
-echo $PWD
-VARS_FILE=.build_vars-`uname -s`.sh
+echo "$PWD"
+VARS_FILE=".build_vars-$(uname -s).sh"
 VARS_FILE2=.build_vars.sh
-if [ -f ./${VARS_FILE} ]; then
-  VARS_LOCAL=./${VARS_FILE}
+if [ -f "./${VARS_FILE}" ]; then
+  VARS_LOCAL="./${VARS_FILE}"
 elif [ -f "../${VARS_FILE}" ]; then
-  VARS_LOCAL=../${VARS_FILE}
-elif [ -f ./${VARS_FILE2} ]; then
-  VARS_LOCAL=./${VARS_FILE2}
+  VARS_LOCAL="../${VARS_FILE}"
+elif [ -f "./${VARS_FILE2}" ]; then
+  VARS_LOCAL="./${VARS_FILE2}"
 elif [ -f "../${VARS_FILE2}" ]; then
-  VARS_LOCAL=../${VARS_FILE2}
+  VARS_LOCAL="../${VARS_FILE2}"
 else
   VARS_LOCAL=""
 fi
 
-if [ -z $VARS_LOCAL ]
+if [ -z "${VARS_LOCAL}" ]
 then
-    echo Build vars file $VARS_FILE does not exist
-    echo Cannot continue
+    echo "Build vars file ${VARS_FILE} does not exist"
+    echo "Cannot continue"
     return 1
 fi
 
-echo Build vars file found: $VARS_LOCAL
-. $VARS_LOCAL
+echo "Build vars file found: ${VARS_LOCAL}"
+. ${VARS_LOCAL}
 
-os=`uname`
+os="$(uname)"
 if [ "$os" = "Darwin" ]; then
     if [ -n "$DYLD_LIBRARY_PATH" ]; then
-	export DYLD_LIBRARY_PATH=${SL_LD_LIBRARY_PATH}:$DYLD_LIBRARY_PATH
+	export DYLD_LIBRARY_PATH=${SL_LD_LIBRARY_PATH}:${DYLD_LIBRARY_PATH}
     else
 	export DYLD_LIBRARY_PATH=${SL_LD_LIBRARY_PATH}
     fi
 fi
 
-if [ -z "$SL_PREFIX" ]
+if [ -z "${SL_PREFIX}" ]
 then
-    SL_PREFIX=`pwd`/install
+    SL_PREFIX="$(pwd)/install"
+fi
+
+function export_pythonpath()
+{
+  PYTHON_VERSION="${1}"
+
+  if [ "${PYTHON_VERSION}" -eq 2 ]; then
+    PYTHONPATH=${SL_PREFIX}/lib64/python2.7/site-packages:${PYTHONPATH}
+  elif [ "${PYTHON_VERSION}" -eq 3 ]; then
+    PYTHONPATH=${SL_PREFIX}/lib64/python3/site-packages:${PYTHONPATH}
+  else
+    echo "unknown Python version: ${PYTHON_VERSION}"
+    return 0
+  fi
+
+  export PYTHONPATH
+}
+
+# look for a valid installation of python
+if [ -x "$(command -v python)" ]; then
+  PYTHON_VERSION="$(python -c 'import sys; print(sys.version_info.major)')"
+  export_pythonpath "${PYTHON_VERSION}"
+# there is still a chance that python3 is installed
+elif [ -x "$(command -v python3)" ]; then
+  PYTHON_VERSION="$(python3 -c 'import sys; print(sys.version_info.major)')"
+  export_pythonpath "${PYTHON_VERSION}"
+else
+  echo "python not found"
 fi
 
 function in_list()
 {
     this=$1
     shift
-    for dir in $*; do
+    for dir in "$@"; do
         if [ "$dir" == "$this" ]; then
             return 1
         fi
     done
     return 0
 }
 
-list=`compgen -A variable | grep "SL_.*_PREFIX"`
+function get_old_path()
+{
+  echo $PATH | sed 's/:/ /g'
```

Comment:
(lint) See if you can use ${variable//search/replace} instead. [SC2001]
(lint) Double quote to prevent globbing and word splitting. [SC2086]